syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.prerepa.gps";
option java_outer_classname = "GpsProto";
option objc_class_prefix = "HLW";

package gps;

// The greeting service definition.
service GpsService {
  // Retrieves coordinates, inserts, sends back a status.
	rpc CaneInsert (CaneCoordinates) returns (InsertStatus) {}
	// Retrieves user and id from database (recieved from app) and sends coordinates.
	rpc AppRetrieve (CoordinatesID) returns (Coordinates) {}
}

// The request message containing the Coordinates of the cane to be inserted.
message CaneCoordinates {
	string username = 1;
	double latitude = 2;
	double longitude = 3;
}

// The response message containing the greetings
message InsertStatus {
	int32 insertStatus = 1;
}

message CoordinatesID {
	string username = 1;
	int32 coordinatesID = 2;
}

message Coordinates {
	double latitude = 1;
	double longitude = 2;
}
